File: Power Query/300-399/392/PQ_392.py
import pandas as pd

path = "300-399/392/PQ_Challenge_392.xlsx"
input = pd.read_excel(path, usecols="A:C", nrows=22, skiprows=0)
test = pd.read_excel(path, usecols="E:I", nrows=3, skiprows=0).rename(
	columns=lambda c: c.replace(".1", "")
)

result = (
	input.sort_values(["User_ID", "Timestamp"])
	.groupby("User_ID", as_index=False)
	.agg(
		Entry_Page=("Page_Visited", "first"),
		Exit_Page=("Page_Visited", "last"),
		Unique_Pages_Visited=("Page_Visited", "nunique"),
		Full_Navigation_Path=("Page_Visited", lambda s: " > ".join(s.astype(str))),
	)
)

print(result.equals(test))
# True

File: Power Query/300-399/392/PQ_392.R
library(tidyverse)
library(readxl)

path <- "300-399/392/PQ_Challenge_392.xlsx"
input <- read_excel(path, range = "A1:C22")
test <- read_excel(path, range = "E1:I4")

result = input %>%
  arrange(User_ID, Timestamp) %>%
  summarise(
    Entry_Page = first(Page_Visited),
    Exit_Page = last(Page_Visited),
    Unique_Pages_Visited = n_distinct(Page_Visited),
    Full_Navigation_Path = paste(Page_Visited, collapse = " > "),
    .by = User_ID
  )

all.equal(result, test)
#> [1] TRUE
